interface_api.py

```diff
@@ -0,0 +1,64 @@
+import http
+import json
+
+from flask import Flask, jsonify, request, Response
+
+
+class InterfaceAPI(Flask):
+    def __init__(self, import_name):
+        super(InterfaceAPI, self).__init__(import_name)
+        self.cache = {}
+        with open('resource/task-1-interfaces.json') as f:
+            data = json.loads(f.read())
+            interfaces = data['ietf-interfaces:interfaces']['interface']
+            for interface in interfaces:
+                self.cache[interface['name']] = interface
+
+
+app = InterfaceAPI(__name__)
+
+
+@app.route('/get-all-interfaces', methods=['GET'])
+def get_all_interfaces() -> Response:
+    all_interfaces = app.cache.values()
+    return jsonify(list(all_interfaces))
+
+
+@app.route('/get-interface/<path:interface_name>', methods=['GET'])
+def get_some_interface(interface_name: str) -> Response | tuple[str, int]:
+    interface = app.cache.get(interface_name)
+    if interface:
+        return jsonify(interface)
+    else:
+        return '', http.HTTPStatus.NOT_FOUND
+
+
+@app.route('/get-interfaces', methods=['POST'])
+def post_several_interfaces() -> Response:
+    all_interfaces = app.cache.values()
+    req_data = request.get_json()
+    result = []
+    for interface in all_interfaces:
+        for input_interface in req_data['input']['interfaces']:
+            interface_name = input_interface.get("name")
+            interface_type = input_interface.get("type")
+            interface_enabled = input_interface.get("enabled")
+            if (interface_name is None or interface["name"] == interface_name) and \
+                    (interface_type is None or interface["type"] == interface_type) and \
+                    (interface_enabled is None or interface["enabled"] == interface_enabled):
+                if interface not in result:
+                    result.append(interface)
```

Comment:
@patrikprovaznik this can be pretty performance heavy but you have to remember that there is almost always something you do when you cycle many times over some things (99.9 percent of time) and usually if you can t then the design of code is not right and should think of redesigning.

my first rule is to never use more then two cycles in each other. you have two which is ok. if you have more it s usually good idea the longest one change to dictionary and try to get data somehow from that instead of looping all the time through all of them.

secondly always looks for some conditions that can break your for loop or continue if you can somehow so only part of cycle will be actually computed and rest will be skipped.

So when you cycling over your interfaces at the end you check if `interface not in result`. which means if you add it in you can break cycle and continue to another interface cause even if it will comply with another condition of your input it will not be added second time anymore. which means that your `if interface not in result` will be useless as well if you break afterwards. same interface will not be in there anymore. This will lower the amount of cycling immensely, especially if first condition will be already true and there is like 10 more options 


It looks like redundant thing in here cause you don t have much of interfaces but usually in projects we have lists of 10 000 things and have similar problem to this